### PR TITLE
Correcting string interpolations failing on julia 1.3

### PR DESCRIPTION
**Pull request by matbesancon**

### CODE REVIEW

**Comment by matbesancon** [comment]
Closes #2042

**Comment by blegat** [comment]
Can you enable `1.3` on Travis and AppVeyor ?

**Comment by matbesancon** [comment]
To go a bit deeper on the original bug, it's not all identifiers, it's just var being broken:
```julia
julia> println("$var.")
ERROR: syntax: unexpected ")"
Stacktrace:
 [1] top-level scope at REPL[3]:0

julia> println("$val.")
2.
```

**Comment by matbesancon** [comment]
Given the answer on https://github.com/JuliaLang/julia/issues/33044, do we still want this?
cc @odow

File: src/parse_nlp.jl [hunk 1]
```diff
@@ -22,7 +22,7 @@ function _parse_NL_expr(m, x, tapevar, parent, values)
         elseif _is_prod(header)
             operatorid = operator_to_id[:*]
         else
-            error("Unrecognized expression $header(...)")
+            error("Unrecognized expression $(header...)")
```
Comment (odow) [inline src/parse_nlp.jl]: Is it meant to be `$(header)(...)`?

File: src/parse_nlp.jl [hunk 2]
```diff
@@ -48,7 +48,7 @@ function _parse_NL_expr(m, x, tapevar, parent, values)
             opname = x.args[1]
             errorstring = "$opname() can appear in nonlinear expressions " *
             " only if the argument is a generator statement, for example, " *
-            "$opname(x[i] for i in 1:N)."
+            "$(opname(x[i] for i in 1:N))."
```
Comment (odow) [inline src/parse_nlp.jl]: I think this is meant to be `$(opname)(x[i] for i in 1:N)`

File: src/parse_nlp.jl (hunk 1 above)
Comment (matbesancon) [reply src/parse_nlp.jl]: fixed thanks

**Comment by c42f** [comment]
Thanks for reporting the `var` bug upstream.

The rule for where string interpolations stop is meant to be super simple: they consume exactly one identifier or parenthesized expression.

**Comment by matbesancon** [comment]
the other errors on 1.3 are about a `_mul!` undefined

**Pull request closed without merging**